hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeDeleteHandler.java

```diff
@@ -42,37 +51,78 @@ public NodeDeleteHandler() {
 
     @Override
     public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
-        // implement
+        final NodeDeleteTransactionBody transactionBody = txn.nodeDeleteOrThrow();
+        final long nodeId = transactionBody.nodeId();
+
+        if (nodeId <= 0 || nodeId > Long.MAX_VALUE) {
+            throw new PreCheckException(INVALID_NODE_ID);
+        }
     }
 
     @Override
     public void preHandle(@NonNull final PreHandleContext context) throws PreCheckException {
         requireNonNull(context);
 
-        final var op = context.body().nodeDelete();
-        throw new UnsupportedOperationException("need implementation");
+        final NodeDeleteTransactionBody transactionBody = context.body().nodeDeleteOrThrow();
+        final ReadableNodeStore nodeStore = context.createStore(ReadableNodeStore.class);
+        final long nodeId = transactionBody.nodeId();
+
+        if (nodeId <= 0 || nodeId > Long.MAX_VALUE) {
+            throw new PreCheckException(INVALID_NODE_ID);
+        }
+
+        Node node = nodeStore.get(nodeId);
+        mustExist(node, INVALID_NODE_ID);
     }
 
     /**
-     * Given the appropriate context, deletes a topic.
+     * Given the appropriate context, deletes a node.
      *
      * @param context the {@link HandleContext} of the active transaction
      * @throws NullPointerException if one of the arguments is {@code null}
      */
     @Override
     public void handle(@NonNull final HandleContext context) {
-        requireNonNull(context, "The argument 'context' must not be null");
+        requireNonNull(context);
+
+        final NodeDeleteTransactionBody transactionBody = context.body().nodeDeleteOrThrow();
+        var nodeId = transactionBody.nodeId();
+
+        if (nodeId <= 0 || nodeId > Long.MAX_VALUE) {
+            throw new HandleException(INVALID_NODE_ID);
+        }
 
-        final var op = context.body().nodeDelete();
-        throw new UnsupportedOperationException("need implementation");
+        final var nodeStore = context.writableStore(WritableNodeStore.class);
+
+        Node node = nodeStore.get(nodeId);
+
+        if (node == null) {
+            throw new HandleException(INVALID_NODE_ID);
+        }
+
+        if (node.deleted()) {
+            throw new HandleException(NODE_DELETED);
+        }
+
+        /* Copy all the fields from existing, and mark deleted flag  */
+        final var nodeBuilder = new Node.Builder()
+                .nodeId(node.nodeId())
+                .accountId(node.accountId())
+                .gossipEndpoint(node.gossipEndpoint())
+                .serviceEndpoint(node.serviceEndpoint())
+                .gossipCaCertificate(node.gossipCaCertificate())
+                .grpcCertificateHash(node.grpcCertificateHash())
+                .weight(node.weight())
+                .deleted(true);
```

Comment:
Can replace the block of code to: nodeBuilder = node.copyBuild().deleted(true).

Reply:
done thanks